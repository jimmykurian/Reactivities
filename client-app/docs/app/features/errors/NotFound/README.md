[**Client App Documentation v0.0.0**](../../../../README.md) â€¢ **Docs**

***

[Client App Documentation v0.0.0](../../../../README.md) / app/features/errors/NotFound

# app/features/errors/NotFound

## Index

### Functions

- [default](functions/default.md)
