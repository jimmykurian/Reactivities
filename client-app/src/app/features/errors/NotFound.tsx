/**
 * @file NotFound.tsx
 * @author Jimmy Kurian
 * @fileoverview This file contains the NotFound component, which displays a user-friendly error message when a page is not found.
 */

import { Link } from 'react-router-dom';
import { Button, Header, Icon, Segment } from 'semantic-ui-react';

/**
 * NotFound component displays a message indicating that the requested page could not be found.
 * It provides a button for users to navigate back to the activities page.
 *
 * @component
 * @returns {JSX.Element} The rendered NotFound component.
 *
 * @example
 * ```tsx
 * import NotFound from './NotFound';
 *
 * function App() {
 *   return (
 *     <BrowserRouter>
 *       <Routes>
 *         <Route path="*" element={<NotFound />} />
 *       </Routes>
 *     </BrowserRouter>
 *   );
 * }
 * ```
 */
export default function NotFound(): JSX.Element {
  return (
    <Segment placeholder>
      <Header icon>
        <Icon name="search" />
        Oops - we&#39;ve looked everywhere and could not find what you are
        looking for!
      </Header>
      <Segment.Inline>
        <Button as={Link} to="/activities" primary>
          Return to activities page
        </Button>
      </Segment.Inline>
    </Segment>
  );
}
