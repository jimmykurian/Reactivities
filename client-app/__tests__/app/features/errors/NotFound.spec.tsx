import { render, screen } from '@testing-library/react';
import NotFound from '../../../../src/app/features/errors/NotFound';
import { BrowserRouter } from 'react-router-dom';

describe('NotFound Component', () => {
  it('should match the snapshot', () => {
    // Arrange
    const { container } = render(
      <BrowserRouter>
        <NotFound />
      </BrowserRouter>,
    );

    // Assert
    expect(container).toMatchSnapshot();
  });

  it('should render the NotFound component with correct content', () => {
    // Arrange
    render(
      <BrowserRouter>
        <NotFound />
      </BrowserRouter>,
    );

    // Assert
    expect(
      screen.getByText(
        "Oops - we've looked everywhere and could not find what you are looking for!",
      ),
    ).toBeInTheDocument();
    expect(
      screen.getByRole('button', { name: /Return to activities page/i }),
    ).toBeInTheDocument();
  });

  it('should have a button that links back to the activities page', () => {
    // Arrange
    render(
      <BrowserRouter>
        <NotFound />
      </BrowserRouter>,
    );

    // Act
    const button = screen.getByRole('button', {
      name: /Return to activities page/i,
    });

    // Assert
    expect(button).toHaveAttribute('href', '/activities');
  });
});
