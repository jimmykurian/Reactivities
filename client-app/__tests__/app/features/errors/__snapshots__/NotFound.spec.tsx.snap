// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Component should match the snapshot 1`] = `
<div>
  <div
    class="ui placeholder segment"
  >
    <div
      class="ui icon header"
    >
      <i
        aria-hidden="true"
        class="search icon"
      />
      Oops - we've looked everywhere and could not find what you are looking for!
    </div>
    <div
      class="inline"
    >
      <a
        class="ui primary button"
        href="/activities"
        role="button"
      >
        Return to activities page
      </a>
    </div>
  </div>
</div>
`;
